### PR TITLE
Documentation: Kconfig - move links to official doc to separate page

### PR DESCRIPTION
**Pull request by asbjornsabo**
Move note with links to official Kconfig documentation to separate
page, to make it visible in the table of contents, so that it is
easier to find.

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
Not sure this small note needs a page of its own, but others may disagree :)
What does the doc change have to do with Bluetooth?

File: doc/guides/build/kconfig/tips.rst [hunk 1]
```diff
@@ -6,13 +6,6 @@ Kconfig - Tips and Best Practices
 This page covers some Kconfig best practices and explains some Kconfig
 behaviors and features that might be cryptic or that are easily overlooked.
 
-.. note::
-
-   The official Kconfig documentation is `kconfig-language.rst
-   <https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html>`__
-   and `kconfig-macro-language.rst
-   <https://www.kernel.org/doc/html/latest/kbuild/kconfig-macro-language.html>`__.
-
```
Comment (stephanosio) [inline doc/guides/build/kconfig/tips.rst:15]: I don't think we should remove the links to the official Kconfig documentation from the "Kconfig - Tips and Best Practices" page because it is, in fact, a very useful tip.

Also the `Bluetooth:` prefix seems to be irrelevant.

Comment (asbjornsabo) [reply doc/guides/build/kconfig/tips.rst:15]: > Not sure this small note needs a page of its own, but others may disagree :)
> What does the doc change have to do with Bluetooth?

1) Yes, the page is small.  The point is to get the pointer to the official docs into the Kconfig doc table of contents, so that they are more visible.  A separate page seems to be the way to do this.
If people do not think this is a good change, I am fine with that.  But it would have helped me.

2) Nothing - will change

**Comment by asbjornsabo** [comment]
> Not sure this small note needs a page of its own, but others may disagree :)
> What does the doc change have to do with Bluetooth?

Yes, the page is small. The point is to get the pointer to the official docs into the Kconfig doc table of contents, so that they are more visible. A separate page seems to be the way to do this.
If people do not think this is a good change, I am fine with that. But it would have helped me.

Nothing - will change

File: doc/guides/build/kconfig/tips.rst (hunk 1 above)
Comment (asbjornsabo) [reply doc/guides/build/kconfig/tips.rst:15]: 1) They are not removed, only moved.  I considered keeping them in the original location as well, but decided against it, as that would duplicate information.  (A certain way to forget to update one of the instances if the links ever change.)
I could add a reference to the new page instead - would that be better?

2) Yes, it is.  It was added by force of habit - it is what I usually write first in a PR subject line.  Will fix.

File: doc/guides/build/kconfig/official_doc.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+Kconfig - Official Documentation
+################################
+
+.. note::
+
+   The official Kconfig documentation is `kconfig-language.rst
+   <https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html>`__
+   and `kconfig-macro-language.rst
+   <https://www.kernel.org/doc/html/latest/kbuild/kconfig-macro-language.html>`__.
```
Comment (gmarull) [inline doc/guides/build/kconfig/official_doc.rst:11]: I'd personally not create an extra page for just a couple of links. Some description with these links in `Configuration System (Kconfig)` section should be enough to provide better visibility.

Comment (asbjornsabo) [reply doc/guides/build/kconfig/official_doc.rst:11]: Better than now, for sure.  But not as good as getting them into the table of contents.

**Review by tejlmand — COMMENTED** [review]
> > Not sure this small note needs a page of its own, but others may disagree :)
> > What does the doc change have to do with Bluetooth?
> 
> Yes, the page is small. The point is to get the pointer to the official docs into the Kconfig doc table of contents, so that they are more visible. A separate page seems to be the way to do this.

I don't think it deserves it own page.
And chapters do get their own entries, no need to create a new page:
image

but personally I don't think we should have a link to Kconfig docs at so high level (as this PR proposes).
That would be like putting a link to CMake or devicetree specification at similar toplevel, which we don't do.

That being said, maybe it would make sense to restructure:
- Links to official Kconfig documentation 
- https://docs.zephyrproject.org/latest/guides/build/kconfig/preprocessor-functions.html
- https://docs.zephyrproject.org/latest/guides/build/kconfig/extensions.html

in such a way that the links to the official docs, and Kconfiglib + Zephyr's extensions to Kconfig are located together.
Any thoughts on this ?

**Comment by asbjornsabo** [comment]
> but personally I don't think we should have a link to Kconfig docs at so high level (as this PR proposes).

The reason I created this PR was to put it that high, as the discussion in Discord showed that at least two persons (myself and one more) had had problems finding the links to the official Kconfig documentation.

Personally, I think it makes some sense.  If one is starting out with e.g. Kconfig, finding documentation for the fundamentals is important before one starts on the "tips and tricks" for advanced usage.

But the PR was intended as a proposal, and this is not my area.  So if the proposal is not good, I am fine to drop the PR.
I'll leave it up for a while though, to see if the discussion can conclude on another solution.

**Comment by tejlmand** [comment]
> > but personally I don't think we should have a link to Kconfig docs at so high level (as this PR proposes).
> 
> The reason I created this PR was to put it that high, as the discussion in Discord showed that at least two persons (myself and one more) had had problems finding the links to the official Kconfig documentation.
> 

Yes, I perfectly understand the reason.
And I think it is a valuable discussion you are raising here :+1: 

But there will always be some part of documentation that is hard to find because it's not on the top-level.
And there will always be someone thinking that exactly that information deserves better highlighting.

That being said, having a general Kconfig introduction with said links is probably better than just moving the links themselves.

And as part of that documentation also describe Zephyr's / Kconfiglib extensions to Kconfig.
Which was why I wrote: `That being said, maybe it would make sense to restructure: .... in such a way that the links to the official docs, and Kconfiglib + Zephyr's extensions to Kconfig are located together.`

> Personally, I think it makes some sense. If one is starting out with e.g. Kconfig, finding documentation for the fundamentals is important before one starts on the "tips and tricks" for advanced usage.

I agree that hiding this under *tips and tricks* is not the best solution.
But if we're going to change, then I would like a broader discussion of how it should look, and not just moving those links.

> 
> But the PR was intended as a proposal, and this is not my area. 

And thank you for that, it's always good to get spotlight on things that can be improved

> So if the proposal is not good, I am fine to drop the PR.
> I'll leave it up for a while though, to see if the discussion can conclude on another solution.

I think your PR has value, but also think it can be improved further.
